### ISSUE TITLE
Decouple questions from catalogs and make catalogs composable

### ISSUE THREAD

**Issue by MyPyDavid** [rdmorganiser/rdmo#450]
## Rationale / Begründung

An Editor would like to be able to compose a new catalog by re-using individual questions of the instance.
This would be helpful for the creation of specific catalogs for joint research projects.

#### Propositions

All questions of a RDMO instance should be listed under /questions/ and be filterable for URI.
A new catalog could be composed by choosing questions and sections from this list.
An alternative option set can be defined so that the questions can be re-used more easily and the copying of a complete catalog is not required.
At the creation of a new catalog, tests can be run to check if all the required attributes cover the views.

## References / Verweise

Issues about the management backend: https://github.com/rdmorganiser/rdmo/issues/361

**Comment by jochenklar** [comment]
In principle this should be the case for all nested elements: Sections, Question sets, Questions, and Options. Need major refactoring though.

**Comment by ikostadi** [comment]
+1 for this feature

**Comment by jochenklar** [comment]
@MyPyDavid `At the creation of a new catalog, tests can be run to check if all the required attributes cover the views.` I think this sentence deserves a new issue. This means that a view should be able to "scan" itself to check if all attributes in this view are present in a given catalog, right? (Or vice versa, a catalog should be able to tell if it can fill a view.)

**Comment by MyPyDavid** [comment]
yes, think this is a separate issue/feature. Is more of a part View and Catalog validation function..
I'll make a new one for it and remove it here.
DE text from this sentence:
> Bei der Erstellung eines Katalogs kann geprüft werden, ob alle notwendigen Attribute für die Ansichten abgedeckt werden, ggf. wird angezeigt, welches Attribut bzw. welche dazu passenden Fragen noch genutzt werden können, um eine bestimmte Ansicht abzudecken.

**Comment by MyPyDavid** [comment]
From discussion:

Conceptually, the `URIs` need to be renewed or redefined.
When `Questions` or `Questionsets` are copied/used in multiple different `Catalogs` they need to keep their own unique ID (now knowns as `URI`) to be able to be used across RDMO instances.
Options:
* change the `key` definition part of the `URI`  to contain eg `Catalog/Page/Section/Questionset` (`A/B/C/D`)
  * need to allow `/` in the `key`
* change the `URIs` to an opaque `UUID`
  * requires a lot of migrations overhead and does not work with existing 
*  allow the `key` or `URI` to be set manually

to be continued..

**Comment by MyPyDavid** [comment]
the elements have now `models.ManyToManyField` with each other

closed by [2.0.0](https://github.com/rdmorganiser/rdmo/releases/tag/2.0.0)
